src/averageAges.js

```diff
@@ -55,9 +67,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const kids = people.filter(kid => (
+    onlyWithSon
+      ? people.find(mother => mother.name === kid.mother)
+        && kid.sex === 'm'
+      : people.find(mother => mother.name === kid.mother)
```

Comment:
i think it`s better to create a variable for all women with children

Reply:
Was fixed